packages/instrument-utils/src/creditCard/CreditCardNumberField/CreditCardNumberField.tsx

```diff
@@ -22,69 +21,80 @@ export interface CreditCardNumberFieldProps {
     name: string;
 }
 
-class CreditCardNumberInput extends PureComponent<FieldProps<string>> {
-    private inputRef: RefObject<HTMLInputElement> = createRef();
-    private nextSelectionEnd = 0;
-
-    componentDidUpdate(): void {
-        if (this.inputRef.current && this.inputRef.current.selectionEnd !== this.nextSelectionEnd) {
-            this.inputRef.current.setSelectionRange(this.nextSelectionEnd, this.nextSelectionEnd);
-        }
-    }
-
-    render(): ReactNode {
-        const { field } = this.props;
-
-        return (
-            <>
-                <TextInput
-                    {...field}
-                    additionalClassName="has-icon"
-                    autoComplete="cc-number"
-                    id={field.name}
-                    onChange={this.handleChange}
-                    ref={this.inputRef}
-                    type="tel"
-                />
-
-                <IconLock />
-            </>
-        );
-    }
-
-    private handleChange: ChangeEventHandler<HTMLInputElement> = (event) => {
-        const separator = ' ';
-        const { value = '' } = event.target;
-        const { field, form } = this.props;
-        const { name, value: previousValue = '' } = field;
-        const selectionEnd = this.inputRef.current && this.inputRef.current.selectionEnd;
-
-        // Only allow digits and spaces
-        if (new RegExp(`[^\\d${separator}]`).test(value)) {
-            return form.setFieldValue(name, previousValue);
-        }
-
-        const maxLength = max(creditCardType(value).map((info) => max(info.lengths)));
-
-        const formattedValue = formatCreditCardNumber(
-            value.replace(new RegExp(separator, 'g'), '').slice(0, maxLength),
-            separator,
-        );
+interface CreditCardNumberInputProps {
+    field: FieldProps<string>['field'];
+    form: FieldProps<string>['form'];
+}
 
-        if (selectionEnd === value.length && value.length < formattedValue.length) {
-            this.nextSelectionEnd = formattedValue.length;
-        } else {
-            this.nextSelectionEnd = selectionEnd || 0;
+const CreditCardNumberInput: FunctionComponent<CreditCardNumberInputProps> = ({
+    field,
+    form,
+}): ReactElement => {
+    const inputRef = useRef<HTMLInputElement>(null);
+    const nextSelectionEndRef = useRef(0);
+
+    useEffect(() => {
+        if (inputRef.current && inputRef.current.selectionEnd !== nextSelectionEndRef.current) {
+            inputRef.current.setSelectionRange(
+                nextSelectionEndRef.current,
+                nextSelectionEndRef.current,
+            );
         }
+    });
```

Comment:
```suggestion
    }, []);
```

Reply:
In theory, `[]` should not be set in order to run this whenever something changes.
And we only trigger updates when `inputRef.current && inputRef.current.selectionEnd !== nextSelectionEndRef.current` is met.